main/src/ca/uwaterloo/flix/api/lsp/provider/SymbolProvider.scala

```diff
@@ -94,9 +94,21 @@ object SymbolProvider {
     * Returns a Method DocumentSymbol from a Sig node.
     */
   private def mkSigDocumentSymbol(s: TypedAst.Sig): DocumentSymbol = s match {
-    case TypedAst.Sig(sym, spec, _) => DocumentSymbol(
-      sym.name, Some(spec.doc.text), SymbolKind.Method, Range.from(sym.loc), Range.from(sym.loc), Nil, Nil,
-    )
+    case TypedAst.Sig(sym, spec, impl) =>
+      val selectionRange = Range.from(sym.loc)
+      val range = impl match {
+        case None => selectionRange
+        case Some(i) => Range.merge(selectionRange, Range.from(i.exp.loc))
```

Comment:
I would this fix would actually be one line.

What happens if you use `impl.exp.loc` or something like that here?

Reply:
It doesn't work then, since the resulting `DocumentSymbol` has a `selectionRange` that is outside `range`.

Reply:
You mean that the two locations don't overlap? I.e. the name is outside the body of the expression?

Reply:
Exactly, I also tried it out and It doesn't work, freezing all other information from being displayed

Reply:
Ok. I now understand the problem at least.

@mlutze Do we need to add a `loc` to `TypedAst.Def` that covers the entire definition? (From the start of the symbol, ideally the def keyword actually, until the end of the expression?)

I think we can do that and this PR becomes 1 line, right?

Reply:
You know, in the interest of fixing the underlying problem, and not just solving this specific PR :)

Reply:
I am missing context here, I don't understand why the current location information is problematic now, It would be strange for me ti have the location of the `sym` for the function taking the whole function definition. Can you explain better why you prefer it that way?

Indeed I was pretty surprise when I found out that for enums and classes the definition was already included and outline info where correctly displayed.

Well however thinking about them make me understand that probably you want to change this to have a common way of storing the information (aka symbols with a "body" include the location of it in the symbol location)

Reply:
I think we should have a new `loc` associated with a whole `def`. Not that we should replace or change any of the existing.

Reply:
I understood but I am missing why

Reply:
Given a signature:

```
class X {

  pub def foo(): Int = 1 + 2

}
```

I would propose that `Sig` (and `Def`) is given an extra location that spans:
```
class X {

  |pub def foo(): Int = 1 + 2|

}
```

while still having both a location for the symbol and a location for the exp.

Does that help?

Reply:
It should span from the first `pub` keyword to the end of the whole thing.